src/Servers/Kestrel/Core/src/Features/ISslStreamFeature.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Security;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Features;
+
+/// <summary>
+/// Feature to set access the TLS application protocol
+/// </summary>
+public interface ISslStreamFeature
+{
+    /// <summary>
+    /// Gets the <see cref="SslStream"/>.
+    /// </summary>
+    SslStream SslStream { get; }
```

Comment:
An http connection returns a null `ISslStreamFeature`? Therefore SslStream is never null?

Reply:
I think you're asking why there's no nullability annotation.  This interface property is just exposing a property that's already on the concrete type, where it's non-nullable.  There's a comment thread above with Chris about when it's actually available, but as far as I've been able to determine, the feature(s) are present if and only if `UseHttps` is called.

Reply:
Right, the feature should only be added if SslStream is available. E.g. don't add it for HTTP/3 or non-TLS.

Reply:
I think that's how it behaves now, but I wanted to confirm that you meant "should" in the normative sense and not in the "you missed something" sense.

Reply:
The code looks correct. I hadn't reviewed it fully when I commented on this.